src/background/background.ts

```diff
@@ -12,11 +12,40 @@ import { vaultMessageHandler } from '@background/vault';
 import { IS_TEST_ENV } from '@common/constants';
 import { CONTENT_SCRIPT_PORT } from '@content-scripts/content-script';
 import { VaultActions } from '@background/vault-types';
-import { ExternalMethods, MessageFromContentScript } from '@content-scripts/message-types';
+import {
+  ExternalMethods,
+  InternalMethods,
+  MessageFromContentScript,
+  MESSAGE_SOURCE,
+} from '@content-scripts/message-types';
+
+function setAppParamOfSourceInstallingExtension() {
+  // The app url is saved for use after onboarding
+  const queryString = window.location.search;
+  const urlParams = new URLSearchParams(queryString);
+  const appUrl = urlParams.get('app');
+  // Left for testing only
+  // const appUrl = 'http://localhost:3000/';
+  if (appUrl) window.localStorage.setItem('appUrl', appUrl);
+}
+
+function getAppParamOfSourceInstallingExtension() {
+  const appUrl = window.localStorage.getItem('appUrl') || undefined;
+  return appUrl;
+}
 
 // Listen for install event
 chrome.runtime.onInstalled.addListener(async details => {
   if (details.reason === 'install' && !IS_TEST_ENV) {
+    setAppParamOfSourceInstallingExtension();
+    const appUrl = getAppParamOfSourceInstallingExtension();
+    // Reload the app tab right after install to detect the extension
+    chrome.tabs.query({ lastFocusedWindow: true }, async tabs => {
+      const appTab = tabs?.find(tab => tab.url === appUrl);
+      if (appTab?.id) {
+        await chrome.tabs.reload(appTab?.id);
+      }
+    });
```

Comment:
what happens when a user might have many tabs of this URL open?

Reply:
Good question, will def have to think on how best to solve that bc will be an issue with new designs too, should we just redirect to the first one it finds?

Reply:
We could also save the tabId and make sure to go back to the tab specifically that launched the download?

Reply:
I personally think we should avoid any redirecting behavior in this PR because it adds a lot of complexity that seems to warrant a larger discussion and scoping around